src/main/java/com/hms_networks/sc/canary/api/CanaryDataPosterThread.java

```diff
@@ -39,14 +40,28 @@ public void run() {
         Logger.LOG_DEBUG("Sending completed payload to Canary");
         CanaryApiResponseStatus requestStatus = CanaryApiRequestSender.processRequest(request);
 
-        if (requestStatus != CanaryApiResponseStatus.GOOD_REQUEST) {
-          Logger.LOG_WARN("Unable to send payload to Canary");
-        } else {
+        if (requestStatus == CanaryApiResponseStatus.GOOD_REQUEST) {
           SessionManager.updateTokenExpiration();
           boolean removed = CanaryDataPayloadManager.removeNextPayload();
           if (!removed) {
             Logger.LOG_WARN("Unable to remove payload from queue");
           }
+        } else if (requestStatus == CanaryApiResponseStatus.ERROR_WAIT_FOR_EXPIRE) {
+          Logger.LOG_WARN(
+              "Waiting for existing sessions to expire before sending more data to Canary");
+          try {
+            final long apiClientTimeoutMillis =
+                SCTimeUnit.SECONDS.toMillis(
+                    CanaryConnectorMain.getConnectorConfig().getApiClientTimeoutSeconds());
+            CanaryConnectorMain.getInstance().setDataPollingBlocked(true);
+            Thread.sleep(apiClientTimeoutMillis);
+            CanaryConnectorMain.getInstance().setDataPollingBlocked(false);
+          } catch (InterruptedException e) {
+            Logger.LOG_SERIOUS("An error occurred while waiting for existing sessions to expire.");
+            Logger.LOG_EXCEPTION(e);

```

Comment:
why not used the combined form here?

 Logger.LOG_SERIOUS("An error occurred while waiting for existing sessions to expire.", e);

Reply:
This update intends to address excessive logging. In the case of the duplicate sessions, there was so much logging that it appeared as if the errors were occurring much more often than they actually were. 

I did consider combined logging here, but in this scenario, the exception did not need to be promoted, as it is not a critical error that prevents the connector from continuing. If an error prevents waiting here, the connector will try another request, possibly triggering the waiting mechanism again. There are some scenarios, though, where this delay ending early would not negatively impact the connector and may even get it in regular/normal operation sooner.

- The wait time is just the connector's best effort, as the data duplication warning is out of the connector's control.
- The wait time was selected as the configured session length so that if a duplicate session (cause of data duplication warning) from the connector exists, it will have expired after the wait. 
- Duplicate sessions between Canary and the connector can exist after an unexpected reboot or crash of the Ewon, which takes time to recover from. This time counts towards expiring any 'orphaned' sessions with Canary, making the delay time longer than required (but necessary to guarantee).